### ISSUE TITLE
Feature Req: Support AZ ID and name on subnet creation

### ISSUE THREAD

**Issue by Ry-K** [terraform-aws-modules/terraform-aws-vpc#234]
We are currently using this module for VPC creation. In order to support alignment across a given set of accounts for an upcoming project, we would like to have the ability to set the Availability Zone ID rather than the name when creating a subnet.

https://www.terraform.io/docs/providers/aws/r/subnet.html#argument-reference

Currently the module only supports name (availability_zone argument) when creating a subnet.

Regards,
Ryan

**Comment by syquus** [comment]
I just landed here looking for how to configure AZ ID on this module. Sad to see it isn't available. It's a must for cross-account compatibility...

**Comment by DrFaust92** [comment]
I'm going to take a crack at this, ill update.
from a glance it seems this will be a breaking change if there is a move from az to az id. it should work fine for new VPCs.

**Comment by bryantbiggs** [comment]
Duplicate of #350

**Comment by antonbabenko** [comment]
I have just released [v2.25.0](https://github.com/terraform-aws-modules/terraform-aws-vpc/releases/v2.25.0) (after #370 has been merged) and you can now specify AZ IDs or AZ names in `azs` argument, like this (pick one):

```hcl
module "vpc" {
  source = "terraform-aws-modules/vpc/aws"

  # Option 1 - Using AZ names
  azs = ["eu-west-1a", "eu-west-1b"]

  # Option 2 - Using AZ ids
  azs = ["euw1-az1", "euw1-az2"]

  # Option 3 - Using mixed values (names and ids)
  azs = ["eu-west-1a", "euw1-az2"]
}
```

Closing this issue.
